src/cascadia/TerminalControl/TermControlAutomationPeer.cpp

```diff
@@ -27,9 +27,11 @@ namespace XamlAutomation
 
 namespace winrt::Microsoft::Terminal::TerminalControl::implementation
 {
-    TermControlAutomationPeer::TermControlAutomationPeer(winrt::Microsoft::Terminal::TerminalControl::implementation::TermControl const& owner) :
-        TermControlAutomationPeerT<TermControlAutomationPeer>(owner), // pass owner to FrameworkElementAutomationPeer
-        _uiaProvider{ owner, std::bind(&TermControlAutomationPeer::GetBoundingRectWrapped, this) } {};
+    TermControlAutomationPeer::TermControlAutomationPeer(winrt::Microsoft::Terminal::TerminalControl::implementation::TermControl* owner) :
+        TermControlAutomationPeerT<TermControlAutomationPeer>(*owner) // pass owner to FrameworkElementAutomationPeer
+    {
+        THROW_IF_FAILED(::Microsoft::WRL::MakeAndInitialize<::Microsoft::Terminal::TermControlUiaProvider>(&_uiaProvider, owner, std::bind(&TermControlAutomationPeer::GetBoundingRectWrapped, this)));

```

Comment:
So, in one of my commits, I tried having this as `LOG_IF_FAILED` instead of `THROW_IF_FAILED` and I'd get a crash on a creating a new tab. Specifically, the `UiaData` extracted from the `TermControl` would be `nullptr`, meaning that we wouldn't be able to really do any operations in the auto peer.

This fixed it by catching the crash. And we're still able to interact with the auto peer well on new tab/pane creation. I just feel lilke I might be missing a better fix. Thoughts?